semcore/d3-chart/src/Dots.jsx

```diff
@@ -76,26 +78,72 @@ function Dots(props) {
     const radius = radiusBase * (active ? 8 / 6 : 1);
     if (!d3.defined()(d)) return acc;
     if (!visible) return acc;
-    acc.push(
-      sstyled(styles)(
-        <SDot
-          aria-hidden
-          key={`${i}`}
-          render='circle'
-          visible={visible}
-          __excludeProps={EXCLUDE_PROPS}
-          value={d}
-          index={i}
-          cx={d3.x()(d)}
-          cy={d3.y()(d)}
-          active={active}
-          hide={hide}
-          color={resolveColor(color)}
-          transparent={transparent}
-          r={radius}
-        />,
-      ),
-    );
+    // acc.push(
+    //   sstyled(styles)(
+    //     <SDot
+    //       aria-hidden
+    //       key={`${i}`}
+    //       render='circle'
+    //       visible={visible}
+    //       __excludeProps={EXCLUDE_PROPS}
+    //       value={d}
+    //       index={i}
+    //       cx={d3.x()(d)}
+    //       cy={d3.y()(d)}
+    //       active={active}
+    //       hide={hide}
+    //       color={resolveColor(color)}
+    //       transparent={transparent}
+    //       radius={radius}
+    //     />,
+    //   ),
+    // );
+    const patternKey = color || getChartDefaultColorName(0);
+    const [width, height] = getPatternSymbolSize({
+      patternKey,
+      patterns,
+    });
+
+    if (!patterns) {
+      acc.push(
+        sstyled(styles)(
+          <SDot
+            render='circle'
+            color={resolveColor(color)}
+            patternKey={patternKey}
+            patterns={patterns}
+            key={`${i}`}
+            visible={visible}
+            active={active}
+            hide={hide}
+            transparent={transparent}
+            cx={d3.x()(d)}
+            cy={d3.y()(d)}
+            r={radius}
+          />,
+        ),
+      );
+    } else {
```

Comment:
maybe it is better to define render as const before `acc.push` and don't use if-else ?

and why do you use `cx`, `cy`, `r` in first Dot and `x`, `y`, `radius` in second. Could we use the same props?

Reply:
> and why do you use cx, cy, r in first Dot and x, y, radius in second. Could we use the same props?

It's the reason why I made if-else instead of single statement with different render prop. 
By default the dot is a circle that may change it's radius. It needs to use the `cx` and `cy` props to center correctly.

In the other case we have PatternSymbol that is a nested svg images. Centering all thing except the circle is a hard process in the svg world. So, it needs `x` and `y` props only and also need to compute width and height offset to render in the right place.